openwisp_firmware_upgrader/templates/admin/firmware_upgrader/batch_upgrade_operation_change_form.html

```diff
@@ -0,0 +1,160 @@
+                </span>
+                <span class="right-arrow">
+                    <img src="{% static "ui/openwisp/images/right-arrow.svg" %}" alt="right">
+                </span>
+                <div aria-label="changelist filters region" tabindex="0" class="ow-filter-slider">
+                    <div class="slider">
+                        {% for spec in filter_specs %}
+                        <div class="ow-filter {{ spec.title|join_string }}">
+                            {% for choice in spec.choices %}
+                                {% if choice.selected %}
+                                <div id="{{ spec.title|join_string }}" aria-label="by {{ spec.title }}"
+                                    tabindex="0" aria-controls="choices-{{ spec.title|join_string }}"
+                                    role="button" aria-expanded="false" class="filter-title">
+                                    <h3>
+                                        {% blocktrans with filter_title=spec.title %} By {{ filter_title }} {% endblocktrans %}
+                                    </h3>
+                                    <div class="selected-option" tabindex="0" title="{{ choice.display }}">
+                                        {{ choice.display }}
+                                    </div>
+                                </div>
+                                {% endif %}
+                            {% endfor %}
+                            <div id="choices-{{ spec.title|join_string }}" role="region"
+                                aria-labelledby="{{ spec.title|join_string }}" class="filter-options">
+                                <ul>
+                                {% for choice in spec.choices %}
+                                    <li>
+                                        <a {% if choice.selected %} class="selected" {% endif %}
+                                            title="{{ choice.display | default_if_none:'------'}}"
+                                            href="{{ choice.query_string|iriencode }}">
+                                            {{ choice.display | default_if_none:'------' }}
+                                        </a>
+                                    </li>
+                                {% endfor %}
+                                </ul>
+                            </div>
+                        </div>
+                        {% endfor %}
+                    </div>
+                </div>
+            </div>
+            <div class="filters-top">
+                <h2>{% trans 'Filter' %}</h2>
+                <div class="filters-control">
+                    {% if has_active_filters %}
+                        <h3 id="changelist-filter-clear" style="margin: 0; font-size: 14px; font-weight: normal;">
+                            <a href="?" style="color: #666; text-decoration: none;">✖ {% trans "Clear all filters" %}</a>
```

Comment:
Why are there inline styles here?
Please use the unicode/html entities for icons.